### PR TITLE
Edit redirect endpoint

### PR DESCRIPTION
**Pull request by monikasulik**

### CODE REVIEW

File: djangocms_versioning/admin.py [hunk 1]
```diff
@@ -134,12 +135,36 @@ def _get_publish_link(self, obj):
...
 
+    def _get_edit_link(self, obj):
+        """Helper function to get the html link to the edit action
+        """
+        if obj.state == PUBLISHED:
+            pks_for_grouper = obj.versionable.for_grouper(
+                obj.grouper).values_list('pk', flat=True)
+            content_type = ContentType.objects.get_for_model(obj.content)
+            drafts = Version.objects.filter(
+                object_id__in=pks_for_grouper, content_type=content_type,
```
Comment (chaosk) [inline djangocms_versioning/admin.py]: `content_type=obj.content_type` and remove line above.

File: djangocms_versioning/admin.py [hunk 2]
```diff
@@ -134,12 +135,36 @@ def _get_publish_link(self, obj):
...
+                state=DRAFT)
+            if drafts.exists():
+                return ''
+        elif not obj.state == DRAFT:
+            # Don't display the link if it can't be archived
```
Comment (chaosk) [inline djangocms_versioning/admin.py]: if it can't be edited

File: djangocms_versioning/admin.py [hunk 3]
```diff
@@ -134,12 +135,36 @@ def _get_publish_link(self, obj):
...
+            return ''
+        edit_url = reverse('admin:{app}_{model}_edit_redirect'.format(
+            app=obj._meta.app_label, model=self.model.__name__.lower(),
```
Comment (chaosk) [inline djangocms_versioning/admin.py]: `model=obj._meta.model_name`

File: djangocms_versioning/cms_toolbars.py [hunk 4]
```diff
@@ -29,43 +25,62 @@ def _is_versioned(self):
...
-            content_type=content_type, object_id=self.toolbar.obj.pk)
-        if version.state != DRAFT:
+        # Add the publish button if in edit mode
+        if self.toolbar.edit_mode_active:
+            item = ButtonList(side=self.toolbar.RIGHT)
+            proxy_model = self._get_proxy_model()
+            publish_url = reverse('admin:{app}_{model}_publish'.format(
+                app=proxy_model._meta.app_label,
+                model=proxy_model.__name__.lower(),
+            ), args=(self.toolbar.obj.pk,))
```
Comment (chaosk) [inline djangocms_versioning/cms_toolbars.py]: This should be a Version pk, not Content pk

File: djangocms_versioning/cms_toolbars.py [hunk 5]
```diff
@@ -29,43 +25,62 @@ def _is_versioned(self):
...
-            disabled=False,
-        )
+        # Add the edit button if in preview mode
+        if self.toolbar.content_mode_active:
+            item = ButtonList(side=self.toolbar.RIGHT)
+            proxy_model = self._get_proxy_model()
+            edit_url = reverse('admin:{app}_{model}_edit_redirect'.format(
+                app=proxy_model._meta.app_label,
+                model=proxy_model.__name__.lower(),
+            ), args=(self.toolbar.obj.pk,))
```
Comment (chaosk) [inline djangocms_versioning/cms_toolbars.py]: This should be a Version pk, not a Content pk

File: djangocms_versioning/admin.py (hunk 3 above)
Comment (monikasulik) [reply djangocms_versioning/admin.py]: If I remember correctly that doesn't work because I need the proxy model here, but I'll try to make sure.

**Pull request closed without merging**